custom/icds_reports/ucr/data_sources/ccs_record_cases_monthly_tableau2.json

```diff
@@ -3110,6 +3110,24 @@
         ]
       }
     },
+    "related_docs": {
+      "person_case_id": {
```

Comment:
these represent ids of indicators inside the config? so they'd be pulled out directly from the row?

Reply:
I imagine this as an expression that calculates the the id from the root document, so it'd look something like https://github.com/dimagi/commcare-hq/blob/5298a6c541c90274cab09154e5e2b00d2b2363a4/custom/ucr_ext/expressions.py#L239-L270

We'd probably want to do something custom if there's any chance to optimize this though so that we can get the child health cases from the ccs record case and vice versa. Right now the hierarchy looks like:
```
                 household
                    /          \
              person      person
              /                     \ 
          ccs_record        child_health
````

There's are some more cases than that. also important to note that a person doesn't always have a child_health or ccs_record and there can be multiple ccs_records and child_health in a household, so we'd need to figure out how to handle possibly N docs from one doc change (this is probably the biggest flag I have here)

Reply:
> there can be multiple ccs_records and child_health in a household, so we'd need to figure out how to handle possibly N docs from one doc change (this is probably the biggest flag I have here)

Is this concern a performance / load one or an implementation one? On the implementation side, if this is a band aid on top of a band aid and the long term plan is to remove both (which I think is what I heard in the meeting) then I imagine we could go as far as to hard code it for each data source if we needed to and just return a list/generator, no? That would be if we didn't have an easy way to return a list of stuff in vanilla UCR (which we probably could as well)

Reply:
It was more of a performance/load one. This makes it harder to reason about the performance since the question of performance is no longer "how long does it take for this case to get this data?", but "how long does it take this case **and it's subcases** to get it's data **and it's subcases data**". And that is complicated because it can be a many-to many relationship (from my understanding of the app).

Another thing that this made me think of is what happens in the following case:

```
                               household 
             /                 /                  \                         \
   person           person               person              person
         /                     /                       \                            \ 
ccs_record          ccs_record        child_health          child_health

```

I don't know if that's a real, valid hierarchy that is expected, but I've seen some test cases like that during QA. So we'd need to recalculate all ccs_records and child_health records under a household to not risk discrepancies. Originally I only planned on evaluating all child_health cases connected to a ccs_record and vice versa, but I think it's possible to have a child_health case not related to the current ccs_record under the household

Reply:
Might be good to quickly sync up over slack/voice if you think this is a concern (and we're moving forward on this). I don't feel that I have a good sense of how this would cause issues. In my head this is just re-ordering work that was all going to happen anyways, but it sounds like there's a concern/constraint you have around knowing how much work you might have to do up front that I don't totally understand. Guessing related to some nuance of how the async indicators function?